src/java.base/share/classes/java/lang/runtime/ObjectMethods.java

```diff
@@ -216,7 +216,17 @@ private static MethodHandle makeEquals(Class<?> receiverClass,
         MethodHandle isInstance = MethodHandles.dropArguments(CLASS_IS_INSTANCE.bindTo(receiverClass), 0, receiverClass); // (RO)Z
         MethodHandle accumulator = MethodHandles.dropArguments(TRUE, 0, receiverClass, receiverClass); // (RR)Z
 
-        for (MethodHandle getter : getters) {
+        var equalsGetters = getters.toArray(new MethodHandle[0]);
+        Arrays.sort(equalsGetters, (mh1, mh2) -> {
+            var rt1 = mh1.type().returnType();
+            var rt2 = mh2.type().returnType();
+            return Integer.compare(
+                rt1.isPrimitive() || rt1.isEnum() || rt1.isArray() ? 1 : Iterable.class.isAssignableFrom(rt1) ? -1 : 0,
```

Comment:
Doesn't this put primitives, enums and arrays at the end instead of at the start? I've tried this with a simple array:
```java
Class<?>[] types = { int.class, String.class, List.class, long.class, TimeUnit.class, byte[].class, Integer.class };
```
The result of sorting:
```
 Class[7] { interface java.util.List, class java.lang.String, class java.lang.Integer, int, long, class java.util.concurrent.TimeUnit, class [B }
```
By switching the -1 and 1 I get the primitives etc.  at the start:
```
 Class[7] { int, long, class java.util.concurrent.TimeUnit, class [B, class java.lang.String, class java.lang.Integer, interface java.util.List }
``

Reply:
The `equalator`s are joined together in reverse order, so this is actually correct for the current implementation:

```java
record Example(A a, B b, C c) {
	public static void main(String... args) {
		final var left	= new Example(new A(), new B(), new C());
		final var right	= new Example(new A(), new B(), new C());

		left.equals(right);
		// prints:
		// > C::equals()
		// > B::equals()
		// > A::equals()
	}
}

record A() {
	@Override
	public boolean equals(Object other) {
		System.out.println("A::equals()");
		return other instanceof A;
	}
}

record B() {
	@Override
	public boolean equals(Object other) {
		System.out.println("B::equals()");
		return other instanceof B;
	}
}

record C() {
	@Override
	public boolean equals(Object other) {
		System.out.println("C::equals()");
		return other instanceof C;
	}
}
```

Reply:
I've added the test case for the current algorithm